tests/docker/ducktape-deps/tinygo-wasi-transforms

```diff
@@ -1,20 +1,39 @@
-#!/bin/sh
+#!/bin/bash
+
 set -e
 set -x
 
+function retry {
+  local retries=$1
+  shift
+
+  local count=0
+  until "$@"; do
+    exit=$?
+    count=$(($count + 1))
+    if [ $count -lt $retries ]; then
+      echo "Retry $count/$retries exited $exit, retrying..."
+    else
+      echo "Retry $count/$retries exited $exit, no more retries left."
+      return $exit
+    fi
+  done
+  return 0
+}
+
 mkdir -p /opt/transforms/tinygo/
 
 cd /transform-sdk/go/transform/internal/testdata
 
-tinygo build -target wasi -opt=z \
+retry 5 tinygo build -target wasi -opt=z \
```

Comment:
is the expectation that if this fails you'll move on to the rest of the tinygo build commands below? if not (i would expect this to be the case), then wouldn't you need `exit $exit` instead of `return $exit`, above?

Reply:
`return $exit` will cause the command to fail and because we use `set -e` that causes the whole script to fail

Reply:
oh, the `set -e` is what i was missing playing with this